app/src/main/java/org/wikipedia/search/SearchResultsFragment.java

```diff
@@ -578,5 +650,88 @@ private Callback callback() {
     private String getSearchLanguageCode() {
         return ((SearchFragment) getParentFragment()).getSearchLanguageCode();
     }
+
+    private void gatherSearchResultsFromAppSources() {
```

Comment:
Whoops sorry missed this @cooltey ... I think whether we are searching cache, prefix or full text we will also need to check our app sources [tabs, lists and history]. Since we only go to full text search after the. title search returns empty, that approach wouldn't work.